x-pack/plugins/security_solution_serverless/server/endpoint/services/set_package_policy_flag.ts

```diff
@@ -60,30 +60,44 @@ async function processBatch(
     return;
   }
 
-  const updatedEndpointPackages = endpointPackagesResult.items.map((endpointPackage) => ({
-    ...endpointPackage,
-    inputs: endpointPackage.inputs.map((input) => {
-      const config = input?.config || {};
-      const policy = config.policy || {};
-      const policyValue = policy?.value || {};
-      const meta = policyValue?.meta || {};
-      return {
-        ...input,
-        config: {
-          ...config,
-          policy: {
-            ...policy,
-            value: {
-              ...policyValue,
-              meta: {
-                ...meta,
-                serverless: true,
+  const updatedEndpointPackages = endpointPackagesResult.items
+    .filter(
+      (endpointPackage) =>
+        !(
+          endpointPackage?.inputs.every(
+            (input) => input.config?.policy?.value?.meta?.serverless ?? false
+          ) ?? false
+        )
+    )
+    .map((endpointPackage) => ({
```

Comment:
nit: using a `reduce` instead of `filter` and `map` would ensure iterating only once over the list instead of twice.